doc/examples/streamline_registration.py

```diff
@@ -0,0 +1,322 @@
+
+"""
+
+affine_reg.level_iters = [1000, 1000, 100]
+affine_map = affine_reg.optimize(static, moving, transform, params0,
+                                 static_affine, moving_affine,
+                                 starting_affine=rigid_map.affine)
+transformed = affine_map.transform(moving)
+
+
+"""
+We now perform the non-rigid deformation using the Symmetric Diffeomorphic
+Registration (SyN) Algorithm proposed by Avants et al. [Avants09]_ (also
+implemented in the ANTs software [Avants11]_):
+
+"""
+
+from dipy.align.imwarp import SymmetricDiffeomorphicRegistration
+from dipy.align.metrics import CCMetric
+
+metric = CCMetric(3)
+level_iters = [10, 10, 5]
+sdr = SymmetricDiffeomorphicRegistration(metric, level_iters)
+
+mapping = sdr.optimize(static, moving, static_affine, moving_affine,
+                       affine_map.affine)
+warped_moving = mapping.transform(moving)
+warped_static = mapping.transform_inverse(moving)
+
+"""
+We show the registration result with:
+
+"""
+from dipy.viz import regtools
+
+regtools.overlay_slices(static, warped_moving, None, 0, "Static", "Moving",
+                        "transformed_sagittal.png")
+regtools.overlay_slices(static, warped_moving, None, 1, "Static", "Moving",
+                        "transformed_coronal.png")
+regtools.overlay_slices(static, warped_moving, None, 2, "Static", "Moving",
+                        "transformed_axial.png")
+
+"""
+.. figure:: transformed_sagittal.png
+   :align: center
+.. figure:: transformed_coronal.png
+   :align: center
+.. figure:: transformed_axial.png
+   :align: center
+
+   Deformable registration result.
+"""
+
+"""
+We read the streamlines from file in voxel space:
+
+"""
+
+from dipy.io.streamline import load_trk
+
+streamlines, hdr = load_trk('lr-superiorfrontal.trk')
+
+
+"""
+We then apply the obtained deformation field to the streamlines. We first
+apply the non-rigid warping and then apply the computed affine transformation
+whose extents must be corrected to account for the different voxel grids of
+the moving and static images:
+
+"""
+
+from dipy.tracking.streamline import deform_streamlines, Streamlines
+
+from dipy.tracking.utils import move_streamlines
+
+warped_streamlines = \
+    deform_streamlines(streamlines,
+                       deform_field=mapping.get_forward_field(),
+                       stream_to_current_grid=moving_affine,
+                       current_grid_to_world=mapping.codomain_grid2world,
+                       stream_to_ref_grid=static_affine,
+                       ref_grid_to_world=mapping.domain_grid2world)
+
+
+adjusted_affine = moving_affine.copy()
+adjusted_affine[0][3] = -adjusted_affine[0][3] - static_affine[0][3]
+adjusted_affine[1][3] = -adjusted_affine[1][3] + static_affine[1][3]
+adjusted_affine[2][3] = -adjusted_affine[2][3] + static_affine[2][3]
+
+rotation, scale = np.linalg.qr(affine_map.affine)
+streams = move_streamlines(streamlines, rotation)
```

Comment:
`transform_streamlines` insted of `move_streamlines`

Reply:
see comment above

Reply:
@dPys' 6974eab commit made it unnecessary using either.